src/core/action/edit/space_domain/ApplyOtsuThresholdEstimationAction.java

```diff
@@ -0,0 +1,132 @@
+package core.action.edit.space_domain;
+
+import core.service.MatrixService;
+import core.service.generation.HistogramService;
+import domain.Histogram;
+import domain.automaticthreshold.OtsuThresholdImage;
+import domain.customimage.CustomImage;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ApplyOtsuThresholdEstimationAction {
+
+    private HistogramService histogramService;
+    private MatrixService matrixService;
+
+    public ApplyOtsuThresholdEstimationAction(HistogramService histogramService, MatrixService matrixService){
+        this.histogramService = histogramService;
+        this.matrixService = matrixService;
+    }
+
+    public OtsuThresholdImage execute(CustomImage customImage){
+        //calculo histograma
+        Histogram histogram = this.histogramService.create(customImage);
+
+        //calculo las sumas acumulativas del histograma
+        Double[] cumulativeProbabilities = new Double[256];
+        for (int i = 0; i <= 255; i++){
+            cumulativeProbabilities[i] = this.cumulativeProbability(histogram, i);
+        }
+
+        //para cada t desde 0 a 255 tengo dos grupos C1 (menor al umbral) y C2 (mayor al umbral)
+
+        //calculo las medias acumulativas (similar a las sumas acumulativas) pero multiplico a pi(p sub i) * i, yendo i de 0 a t
+        Double[] cumulativeMeans = new Double[256];
+        for (int t = 0; t <= 255; t++){
```

Comment:
te parece si juntamos estos dos for?

Reply:
podriamos aprovechar para sacarlo a un metodo tambien

Reply:
Si lo sacamos a un método no podríamos juntar los for. Lo puse así para que quede bien claro el paso a paso, porque es medio jodido de entender otsu.

Reply:
Está implementado donde se ecualiza la imagen. Y vas a ver que es muy distinto al que implementé en Otsu. Por eso no lo implementé. 
Y por otro lado a lo que me refería con no poder juntar los for, sería que estarías poniendo dos responsabilidades a un mismo método, por eso me refería a que no los podes juntar.